pkg/steps/pod_test.go

```diff
@@ -71,7 +93,13 @@ func preparePodStep(t *testing.T, namespace string) (*podStep, stepExpectation,
 	return ps.(*podStep), specification, client
 }
 
-func makeExpectedPod(step *podStep, phaseAfterRun v1.PodPhase) *v1.Pod {
+func makeExpectedPod(step *podStep, phaseAfterRun v1.PodPhase, t *testing.T) *v1.Pod {
```

Comment:
So in general there is a good reason not to have helpers like this. What are we testing? I know you're new to this code but in general we should prefer hard-coded literals for test structures. When we confound the testing with large helpers, we lose the simplicity of input/output checking. Furthermore, you are using `json.Marshal` as well as `decorate.KubeEnv` in the creation of the environment -- the test should not care. These are implementation details today of how these values are created. Your test should validate that the output `PodSpec` is correct.